backend/src/models/db/init_sql.js

```diff
@@ -151,3 +156,125 @@ function sqlCourse(course) {
+
+        for (let i = 0; i < numComments; i++) {
+            const index = getRandomIntInclusive(0, numCommentTypes - 1)
+            const uid = getRandomIntInclusive(1, NUM_DUMMY_USERS)
+            const comment = {
+                ...parent,
+                commentParent: 1,
+                userID: uid,
+                ...commentTypes[index]
+            }
+            comments.push(comment)
+        }
+    }
+
+    return bulkInsertDB('comment', comments)
+}
+
+function sqlUsers() {
+    const userNames = SAMPLE_USERS
+    const suffixes = ['XxX', '!', 's', '!!', '_', '__', 'x']
+
+    let users = []
+
+    for (let i = 1; i < NUM_DUMMY_USERS; i++) {
+        const uid = 'userID' + i
+        const displayName =
+            userNames[i % userNames.length] +
+            // only append a number if we've run out of names
+            (i < userNames.length
+                ? ''
+                // then choose between a simulated birth year and a 'cool' suffix
+                : (i % 2
+                    ? (90 + Math.trunc(i / userNames.length))
+                    // only add a number on the suffix if we're past possible combinations without numbers..
+                    // multiply i by 3 to make it look like a birthdate or something
+                    : (suffixes[i % suffixes.length] + (i < (userNames.length + suffixes.length * 2) ? '' : i * 2))
+                )
+            )
+        const email = displayName + '@test.com.au'
+        const degree = degreeData[getRandomIntInclusive(0, degreeData.length - 1)].name
+
+        users.push({
+            id: i,
+            uid: uid,
+            displayName: displayName,
+            email: email,
+            degree: degree,
+            reputation: userRepMap[i]
+        })
+    }
+
+    return bulkInsertDB('user', users)
+}
+
+function sqlLikes(parents) {
+    let likes = []
+
+    for (let parent of parents) {
+        const numLikes = getRandomIntInclusive(-2, 10)
+        if (numLikes <= 0) continue
+        // choose numLikes consecutive users for these likes...
+        const startIndex = getRandomIntInclusive(1, NUM_DUMMY_USERS)
+
+        for (let i = startIndex; i < startIndex + numLikes; ++i) {
+            const like = {
+                userID: (i % NUM_DUMMY_USERS) + 1,
+                // more likely to be positive!
+                value: Math.random() > 0.7 ? -1 : 1,
+                objectType: parent.objectType,
+                objectID: parent.objectID
+            }
+            // update user reputation for the liked object's user, to be used in initUserTable
+            if (parent.userID in userRepMap) {
+                userRepMap[parent.userID] += like.value
+            } else {
+                userRepMap[parent.userID] = like.value
+            }
+            likes.push(like)
+        }
+    }
+
+    return bulkInsertDB('likes', likes)
+}
+
+/* SQL HELPERS */
+/* THESE ARE ONLY USED FOR TESTING AND ARE VULNERABLE TO SQL INJECTION */
```

Comment:
I don't think we need these at all, it should just work the same way the other sql is generated.

Reply:
So select gets run cause I was too lazy to figure out how to get the returned output from running a shell script, bulkInsert is just a wrapper that returns some SQL (doesn't run it) and runSQL is a wrapper around your method. The reason I chose to uses these is because items with dependencies can be more easily generated, instead of hardcoding queries with assumed ids of things that don't yet exist. This means it should be relatively easy to adapt our testing and get more variety in our test cases. The cost is `~5 seconds`.

Reply:
Is the error with travis tests failing related to this though? I'm not not convinced. Understand what you're saying. Also this is for development, so let's not sweat too much ;-/.

Reply:
Fixed